include/dd/Package.hpp

```diff
@@ -485,6 +485,80 @@ template <class Config> class Package {
     return f;
   }
 
+  // generate general GHZ state with n qubits
+  vEdge makeGHZState(const std::size_t n) {
+    if (n > nqubits) {
+      throw std::runtime_error{
+          "Requested state with " + std::to_string(n) +
+          " qubits, but current package configuration only supports up to " +
+          std::to_string(nqubits) +
+          " qubits. Please allocate a larger package instance."};
+    }
+
+    if (n == 0) {
+      throw std::runtime_error{
+          "Cannot create GHZ state for 0 qubits. Please choose n > 0."};
+    }
+
+    auto leftSubtree = vEdge::one();
+    auto rightSubtree = vEdge::one();
+
+    for (std::size_t p = 0; p < n - 1; ++p) {
+      leftSubtree = makeDDNode(static_cast<Qubit>(p),
+                               std::array{leftSubtree, vEdge::zero()});
+      rightSubtree = makeDDNode(static_cast<Qubit>(p),
+                                std::array{vEdge::zero(), rightSubtree});
+    }
+
+    auto f = makeDDNode(static_cast<Qubit>(n - 1),
+                        std::array<vEdge, RADIX>{
+                            {{leftSubtree.p, cn.lookup(dd::SQRT2_2, 0)},
+                             {rightSubtree.p, cn.lookup(dd::SQRT2_2, 0)}}});
+
+    return f;
+  }
+
+  // generate general W state with n qubits
+  vEdge makeWState(const std::size_t n) {
+    if (n > nqubits) {
+      throw std::runtime_error{
+          "Requested state with " + std::to_string(n) +
+          " qubits, but current package configuration only supports up to " +
+          std::to_string(nqubits) +
+          " qubits. Please allocate a larger package instance."};
+    }
+
+    if (n == 0) {
+      throw std::runtime_error{
+          "Cannot create W state for 0 qubits. Please choose n > 0."};
+    }
+
+    if (n == 1) {
+      return makeBasisState(n, {BasisStates::one});
+    }
+
+    auto leftSubtree = vEdge::one();
+    auto rightSubtree = vEdge::one();
+
+    leftSubtree = makeDDNode(static_cast<Qubit>(0),
+                             std::array{vEdge::zero(), leftSubtree});
+    rightSubtree = makeDDNode(static_cast<Qubit>(0),
+                              std::array{rightSubtree, vEdge::zero()});
+
+    for (size_t p = 1; p < n - 1; ++p) {
+      leftSubtree = makeDDNode(static_cast<Qubit>(p),
+                               std::array{leftSubtree, rightSubtree});
+      rightSubtree = makeDDNode(static_cast<Qubit>(p),
+                                std::array{rightSubtree, vEdge::zero()});
+    }
+
+    auto f = makeDDNode(static_cast<Qubit>(n - 1),
+                        std::array{leftSubtree, rightSubtree});
+    f = normalize(f, false);
+
+    return f;
```

Comment:
First of all: I am really surprised that the logic here is so simple. Wouldn't have expected that!
I even think you can simplify that further; see the suggestion below.
Lastly, the final normalisation call should not be necessary as every node coming from `makeDDNode` should be normalised. Is the top edge weight of the DD correctly set to 1 here? I would have expected some kind of modification to be necessary.

```suggestion
    auto leftSubtree = vEdge::zero();
    auto rightSubtree = vEdge::one();
    for (size_t p = 0; p < n - 1; ++p) {
      leftSubtree = makeDDNode(static_cast<Qubit>(p),
                               std::array{leftSubtree, rightSubtree});
      rightSubtree = makeDDNode(static_cast<Qubit>(p),
                                std::array{rightSubtree, vEdge::zero()});
    }
    return makeDDNode(static_cast<Qubit>(n - 1),
                        std::array{leftSubtree, rightSubtree});
```